### PR TITLE
fix(style): Update search bar and apply styles

### PR DESCRIPTION
**Pull request by Bouncey**
#### Pre-Submission Checklist

- [x] Your pull request targets the `staging` branch of freeCodeCamp.
- [x] Branch starts with either `fix/`, `feature/`, or `translate/` (e.g. `fix/signin-issue`)
- [x] You have only one commit (if not, [squash](http://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git/13231) them into one commit).
- [x] All new and existing tests pass the command `npm test`. Use `git commit --amend` to amend any fixes.

#### Type of Change

- [ ] Small bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Add new translation (feature adding new translations)

#### Checklist:

- [x] Tested changes locally.
- [x] Addressed currently open issue (replace XXXXX with an issue no in next line)

Closes #XXXXX

#### Description

This update brings fuzzy-full-text search to freeCodecamp. The results show highlighted matches, and we now have a custom placeholder.

animation of the results being shown from the search bar

### CODE REVIEW

**Comment by QuincyLarson** [comment]
@Bouncey This looks much better. Can you turn off auto-complete for that form field? The browser pops up a recommendation before the results can render

File: common/app/Nav/Nav.jsx [hunk 1]
```diff
@@ -204,7 +204,10 @@ export class FCCNav extends React.Component {
               />
             </a>
           </NavbarBrand>
-          <FCCSearchBar dropdown={ true } />
+          <FCCSearchBar
+            dropdown={ true }
+            placeholder='&#xf002; What would you like to know?'
```
Comment (raisedadead) [inline common/app/Nav/Nav.jsx]: Consider using a FA icon here instead.

**Review by raisedadead — DISMISSED** [review]

**Comment by raisedadead** [comment]
I tried a search `headline` and continued to `headline wit` which did not seem to yeild a result.
That is I left the next word incomplete.

**Comment by raisedadead** [comment]
BTW, @Bouncey nice work on the component.

**Comment by raisedadead** [comment]
I have more notes, but I'll create a separate issue for them, so that we are able to make separate small PR, and have a MVP.

File: common/app/Nav/Nav.jsx (hunk 1 above)
Comment (Bouncey) [reply common/app/Nav/Nav.jsx]: Believe me, this is the 'least' hacky way I could get a FA icon in the placeholder, other methods require crazy css placement over the input, and there was never a guarantee that the icon would be where we needed it.

This is no HMTL allowed in a placeholder, unfortunately.

**Comment by Bouncey** [comment]
> I tried a search headline and continued to headline wit which did not seem to yeild a result.
That is I left the next word incomplete.

This is a learning curve I am on with elasticsearch. I tried making a query against the title fuzzy, but i think I will have to refactor this. The querying is done [over here](https://github.com/freeCodeCamp/search/blob/cf00e5170aa34fa44a166e7a9ec6a5b025618011/elastic/index.js#L119).

I also want to make contributing the freecodecamp/search more friendly, at the minute there is quite alot of overhead just to get setup locally. I'm thinking a docker-like solution would be a good fit.

**Comment by Bouncey** [comment]
Autocomplete of now off

anim

**Pull request merged**